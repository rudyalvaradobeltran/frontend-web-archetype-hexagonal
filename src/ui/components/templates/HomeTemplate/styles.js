import styled from 'styled-components';

export const Wrapper = styled.div`
  display: grid;
  grid-template-columns: minmax(250px, 780px);
  border: 2px solid #767373;
  justify-content: center;
  background-color: #cbcbcb;
`;

export const UsersGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
`;

export default Wrapper;
