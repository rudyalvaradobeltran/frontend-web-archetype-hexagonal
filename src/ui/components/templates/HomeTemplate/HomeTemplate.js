import React from 'react';
import PropTypes from 'prop-types';

import AvatarUser from '../../atoms/AvatarUser/AvatarUser';

import { Wrapper, UsersGrid } from './styles';

const HomeTemplate = ({ users }) => (
  <Wrapper>
    {users.length > 0 && (
      <UsersGrid>
        {users.map((u) => (
          <AvatarUser key={u.id} name={u.name} phone={u.phone} />
        ))}
      </UsersGrid>
    )}
  </Wrapper>
);

HomeTemplate.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string, phone: PropTypes.string })),
};
HomeTemplate.defaultProps = { users: [] };

export default HomeTemplate;
