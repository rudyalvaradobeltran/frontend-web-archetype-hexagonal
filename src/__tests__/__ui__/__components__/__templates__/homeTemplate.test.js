import React from 'react';
import { create } from 'react-test-renderer';

// Mocks
import MockProvider from '../../../../__mocks__/mockProvider';
// Components
import HomeTemplate from '../../../../ui/components/templates/HomeTemplate/HomeTemplate';

describe('HOME TEMPLATE', () => {
  const component = (children) => <MockProvider>{children}</MockProvider>;

  it('Should render correctly (snapshot) with default props', () => {
    const wrapper = create(component(<HomeTemplate />));
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Should renders correctly (snapshot) whit  props', () => {
    const wrapper = create(
      component(<HomeTemplate users={[{ id: 1, name: 'Jhon Doe', phone: '8875547' }]} />)
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
