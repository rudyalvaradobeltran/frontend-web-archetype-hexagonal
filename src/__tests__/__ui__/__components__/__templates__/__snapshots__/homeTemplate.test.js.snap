// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AVATAR USER Should render correctly (snapshot) with default props 1`] = `
<div
  className="sc-gsDJrp hjRpwD"
/>
`;

exports[`AVATAR USER Should renders correctly (snapshot) whit  props 1`] = `
<div
  className="sc-gsDJrp hjRpwD"
/>
`;

exports[`HOME TEMPLATE Should render correctly (snapshot) with default props 1`] = `
<div
  className="sc-gsDJrp hjRpwD"
/>
`;

exports[`HOME TEMPLATE Should renders correctly (snapshot) whit  props 1`] = `
<div
  className="sc-gsDJrp hjRpwD"
>
  <div
    className="sc-dkPtyc jnbgDF"
  >
    <div
      className="sc-bdvvaa bBZNia"
    >
      <img
        alt="avatar_img "
        src="https://miro.medium.com/max/360/1*W35QUSvGpcLuxPo3SRTH4w.png"
      />
      <p>
        Jhon Doe
      </p>
      <p>
        8875547
      </p>
    </div>
  </div>
</div>
`;
